### DISCUSSION TITLE
Wrapping Arithmetic Operators for Integers: +~, -~, *~ or /~

### DISCUSSION THREAD

**Discussion by igotfr** [vlang/v#11921] [Ideas]
Wrapping Operations for Integers
These operations have guaranteed wraparound semantics. Failing when the result cause overflow

- +~ (wraparound addition)
- -~ (wraparound subtraction)
- *~ (wraparound negation)
- /~ (wraparound multiplication)

Example:
```v
u8(129) +~ u8(129) // error
```

**Comment by Cons-Cat** [comment]


**Reply by igotfr** [reply]
Rust has the checked trait: https://docs.rs/num-traits/0.2.6/num_traits/ops/checked/index.html

**Comment by igotfr** [comment]
related:
https://github.com/vlang/rfcs/issues/10
